### PR TITLE
optimize: publish images based on java `8`, `8-slim`, `17`, `17-slim`

### PR DESCRIPTION
**Pull request by wangliang181230**
optimize: Publish images to `Docker Hub` based on java `8`, `8-slim`, `17`, `17-slim`, and upgrade `setup-java` to `3.9.0`, and simplify the action script.

1. 基于 `java8`, `java8-slim`, `java17`, `java17-slim` 发布docker镜像到 `Docker Hub` 上；
2. 升级 `setup-java` 至 `3.9.0`，并简化发布脚本。

另外，调整 `build.yml` 的脚本：
1. `compile` 的job，编译java11，同时运行 `checkstyle` .
2. `build` 的job，测试java8和17。

---

**参照我fork仓库里的action记录，执行成功了。**
https://github.com/wangliang181230/seata/actions/runs/3779671964

---

**我自己的测试仓库里，上传成功的镜像记录截图。**
https://hub.docker.com/r/easyj/seata-server/tags
图片

### CODE REVIEW

**Review by a364176773 — APPROVED** [review]
LGTM

File: console/pom.xml [hunk 1]
```diff
@@ -35,6 +35,15 @@
...
     <dependencyManagement>
         <dependencies>
+            <!-- junit5 -->
+            <dependency>
+                <groupId>org.junit</groupId>
+                <artifactId>junit-bom</artifactId>
+                <version>${junit-jupiter.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
```
Comment (wangliang181230) [inline console/pom.xml:45]: 避免项目下载低版本的junit5的依赖。

File: .github/workflows/publishes.yml [hunk 2]
```diff
@@ -10,48 +10,73 @@ on:
...
           server-id: oss_seata
-          server-username: OSSRH_USERNAME
-          server-password: OSSRH_PASSWORD
-
-      # step 3 for type=OSSRH
-      - name: "Import GPG-SECRET-KEY, Package and Publish-To-OSSRH"
-        if: matrix.type == 'OSSRH'
+          server-username: OSSRH_USERNAME # Environment variable name for the username for authentication to the Apache Maven repository. Default is $GITHUB_ACTOR
+          server-password: OSSRH_PASSWORD # Environment variable name for password or token for authentication to the Apache Maven repository. Default is $GITHUB_TOKEN
+          gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
```
Comment (slievrly) [inline .github/workflows/publishes.yml:32]: Why not declare it under env?

File: .github/workflows/publishes.yml [hunk 3]
```diff
@@ -10,48 +10,73 @@ on:
...
+            ./mvnw -T 4C clean package -Dimage.name=openjdk:8u342      -Pimage                                    -DskipTests -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
           else
-            ./mvnw -T 4C clean package -Pimage,release-image -DskipTests -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
+            ./mvnw -T 4C clean package -Dimage.name=openjdk:8u342      -Pimage,release-image-based-on-java8       -DskipTests -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
+            ./mvnw -T 4C clean package -Dimage.name=openjdk:8u342-slim -Pimage,release-image-based-on-java8-slim  -DskipTests -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
           fi
+      # step 3 based on java17
+      - name: "Publish images to DockerHub based on java17"
+        if: ${{ matrix.java == 17 && github.ref_name != 'develop' && github.ref_name != 'snapshot' }}
```
Comment (slievrly) [inline .github/workflows/publishes.yml:76]: why include 2.x?

File: .github/workflows/publishes.yml [hunk 4]
```diff
@@ -10,48 +10,73 @@ on:
...
+      - name: "Publish images to DockerHub based on java8"
+        if: matrix.java == 8
         env:
-          IMAGE_NAME: openjdk:8u342
           REGISTRY_USERNAME: ${{ secrets.REGISTRY_USERNAME }}
           REGISTRY_PASSWORD: ${{ secrets.REGISTRY_PASSWORD }}
         run: |
           if [ "${{ github.ref_name }}" == "develop" ] || [ "${{ github.ref_name }}" == "snapshot"  || [ "${{ github.ref_name }}" == "2.x" ]; then
-            ./mvnw -T 4C clean package -Pimage               -DskipTests -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
+            ./mvnw -T 4C clean package -Dimage.name=openjdk:8u342      -Pimage                                    -DskipTests -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
```
Comment (slievrly) [inline .github/workflows/publishes.yml:69]: Is the default "latest" when image.tags is not declared?

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -7,53 +7,74 @@ on:
...
+          java-version: ${{ matrix.java }}
+      # step 3
+      - name: "Build with Maven"
         run: |
-               docker run --rm -v ${{ github.workspace }}:/ws:rw --workdir=/ws \
-               arm64v8/ubuntu:20.04 \
-               bash -exc 'apt-get update -y && \
-                apt-get install maven -y && \
-                mvn -Prelease-seata -Dmaven.test.skip=true clean install -U'
+          ./mvnw -T 4C clean test-compile -Dcheckstyle.skip=false -Dlicense.skip=false -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
```
Comment (slievrly) [inline .github/workflows/build.yml:31]: just compile  src under`src/test`  ?

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -7,53 +7,74 @@ on:
     branches: [ 2.x, develop, master ]
 
 jobs:
-  build_arm64-binary: 
-     runs-on: ubuntu-latest
-     if: ${{ github.event_name == 'push' && (github.ref == 'refs/heads/develop' || github.ref == 'refs/heads/2.x') }}
-     strategy:
+  # job 1: Compile based on java11, and checkstyle.
+  compile:
+    name: "compile"
```
Comment (slievrly) [inline .github/workflows/build.yml:12]: I think the hierarchy is clearer and simpler without splitting(`compile` and  `build` )

File: .github/workflows/build.yml (hunk 5 above)
Comment (wangliang181230) [reply .github/workflows/build.yml:31]: No, it contains `src/main`

File: .github/workflows/publishes.yml (hunk 2 above)
Comment (wangliang181230) [reply .github/workflows/publishes.yml:32]: `gpg-private-key` is the properties of the `actions/setup-java@v3.9.0`. It's the value of the `gpg-private-key`, not an environment variable name.

File: .github/workflows/publishes.yml (hunk 4 above)
Comment (wangliang181230) [reply .github/workflows/publishes.yml:69]: `<image.tags>latest</image.tags>` is configured in the global

File: .github/workflows/publishes.yml (hunk 3 above)
Comment (wangliang181230) [reply .github/workflows/publishes.yml:76]: done

File: .github/workflows/build.yml (hunk 6 above)
Comment (wangliang181230) [reply .github/workflows/build.yml]: done, remove the `compile` job.

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**